src/App.js

```diff
@@ -27,14 +36,48 @@ const MemeContainer = styled(Flex)`
   position: relative;
 `;
 
+const DEFAULT_IMG_URL = '';
+
+const bypassCorsUrl = url => `https://meme.st6.io/meme-image?imageUrl=${url}`;
+
 const App = () => {
   const isMobile = useMedia({ maxWidth: '40em' });
   const [theme, setTheme] = useState('light');
+  const [showUrlInput, setShowUrlInput] = useState(false);
   const [topLabel, setTopLabel] = useState('Do the most meaningful meme...');
   const [bottomLabel, setBottomLabel] = useState('...of your life');
+  const [imageUrl, setImageUrl] = useState(DEFAULT_IMG_URL);
   const [imageSrc, setImageSrc] = useState(getRandomMeme());
   const ref = useRef(null);
 
+  const hideUrlInput = useCallback(() => {
```

Comment:
Do you need `useCallback` as there are no dependencies? This can be just a function that calls the two setters.

Reply:
because we wouldn't recreate a new function on each render, otherwise this would cause the `Meme` component to re-render which on its behalf would lead to invoking hideUrlInput because the `onload` image would be raised.